sdk/cosmos/sdk/src/main/java/com/azure/data/cosmos/sync/CosmosSyncContainer.java

```diff
@@ -0,0 +1,263 @@
+     * @return the cosmos sync item response
+     * @throws CosmosClientException the cosmos client exception
+     */
+    CosmosSyncItemResponse mapItemResponseAndBlock(Mono<CosmosItemResponse> itemMono)
+            throws CosmosClientException {
+        try {
+            return itemMono
+                           .map(this::convertResponse)
+                           .block();
+        } catch (Exception ex) {
+            final Throwable throwable = Exceptions.unwrap(ex);
+            if (throwable instanceof CosmosClientException) {
+                throw (CosmosClientException) throwable;
+            } else {
+                throw ex;
+            }
+        }
+    }
+
+    /**
+     * Read all items iterator.
+     *
+     * @param options the options
+     * @return the iterator
+     */
+    public Iterator<FeedResponse<CosmosItemProperties>> readAllItems(FeedOptions options) {
+        return getFeedIterator(this.containerWrapper.readAllItems(options));
+    }
+
+    /**
+     * Query items iterator.
+     *
+     * @param query the query
+     * @param options the options
+     * @return the iterator
+     */
+    public Iterator<FeedResponse<CosmosItemProperties>> queryItems(String query, FeedOptions options) {
+        return getFeedIterator(this.containerWrapper.queryItems(query, options));
+    }
+
+    /**
+     * Query items iterator.
+     *
+     * @param querySpec the query spec
+     * @param options the options
+     * @return the iterator
+     */
+    public Iterator<FeedResponse<CosmosItemProperties>> queryItems(SqlQuerySpec querySpec, FeedOptions options) {
+        return getFeedIterator(this.containerWrapper.queryItems(querySpec, options));
+    }
+
+    /**
+     * Query change feed items iterator.
+     *
+     * @param changeFeedOptions the change feed options
+     * @return the iterator
+     */
+    public Iterator<FeedResponse<CosmosItemProperties>> queryChangeFeedItems(ChangeFeedOptions changeFeedOptions) {
+        return getFeedIterator(this.containerWrapper.queryChangeFeedItems(changeFeedOptions));
+    }
+
+    /**
+     * Gets item.
+     *
+     * @param id the id
+     * @param partitionKey the partition key
+     * @return the item
+     */
+    public CosmosSyncItem getItem(String id, Object partitionKey) {
+        return new CosmosSyncItem(id,
+                partitionKey,
+                this,
+                containerWrapper.getItem(id, partitionKey));
+    }
+
+    /**
+     * Convert response cosmos sync item response.
+     *
+     * @param response the cosmos item response
+     * @return the cosmos sync item response
+     */
+    CosmosSyncItemResponse convertResponse(CosmosItemResponse response) {
+        return new CosmosSyncItemResponse(response, null, this);
+    }
+
+    private Iterator<FeedResponse<CosmosItemProperties>> getFeedIterator(Flux<FeedResponse<CosmosItemProperties>> itemFlux) {
+        return itemFlux.toIterable().iterator();
```

Comment:
we should control the batch size in converting an observable to Iterator at least for items:
Flux.toIterator(1)

https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#toIterable-int-

Reply:
Added the batch size to items feed